labgrid/remote/client.py

```diff
@@ -1795,8 +1794,32 @@ def main():
         try:
             signal.signal(signal.SIGTERM, lambda *_: sys.exit(0))
 
-            session = start_session(args.crossbar, os.environ.get("LG_CROSSBAR_REALM", "realm1"),
-                                    extra)
+            realm = None
+            if env is not None:
+                # Try getting crossbar URL and realm from config file
+                if args.crossbar is None:
+                    try:
+                        args.crossbar = env.config.get_option('crossbar_url', None)
+                    except KeyError:
+                        pass
+                try:
+                    realm = env.config.get_option('crossbar_realm', None)
+                except KeyError:
+                    pass
+
+            if args.crossbar is None:
+                # If not set, get crossbar URL from LG_CROSSBAR environment variable
+                # fallback to default
+                args.crossbar = os.environ.get("LG_CROSSBAR", "ws://127.0.0.1:20408/ws")
+
+            if realm is None:
+                # If not set, get crossbar realm from LG_CROSSBAR_REALM environment variable
+                # fallback to default
+                realm = os.environ.get("LG_CROSSBAR_REALM", "realm1")
+
+            if args.verbose:
+                print(f"Starting session to crossbar (url '{args.crossbar}' realm '{realm}')")
+            session = start_session(args.crossbar, realm, extra)
```

Comment:
This can be simplified to:
```suggestion
            try:
                crossbar_url = args.crossbar or env.config.get_option('crossbar_url')
            except (AttributeError, KeyError):
                # in case of no env or not set, use LG_CROSSBAR env variable or default
                crossbar_url = os.environ.get("LG_CROSSBAR", "ws://127.0.0.1:20408/ws")

            try:
                crossbar_realm = env.config.get_option('crossbar_realm')
            except (AttributeError, KeyError):
                # in case of no env or not set, use LG_CROSSBAR_REALM env variable or default
                crossbar_realm = os.environ.get("LG_CROSSBAR_REALM", "realm1")

            logging.debug('Starting crossbar session with "%s", realm: "%s")', crossbar_url,
                          crossbar_realm)
            session = start_session(crossbar_url, crossbar_realm, extra)
```

Reply:
Right, it's simpler that way.